test/integration/cni-egress/pod_egress_suite_test.go

```diff
@@ -0,0 +1,138 @@
+// permissions and limitations under the License.
+
+package cni_egress
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework"
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	InstanceTypeNodeLabelKey = "beta.kubernetes.io/instance-type"
+	DEFAULT_VETH_PREFIX      = "eni"
+	DEFAULT_MTU_VAL          = "9001"
+)
+
+var f *framework.Framework
+var maxIPPerInterface int
+var primaryNode v1.Node
+
+// var primaryInstance *ec2.Instance
+var primaryNodeIp *string
+
+// var instanceSecurityGroupID string
+var vpcCIDRs []string
+var isIPv4Cluster bool
+
+func TestCNIPodNetworking(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "CNI Pod Networking Suite")
+}
+
+var _ = BeforeSuite(func() {
+	f = framework.New(framework.GlobalOptions)
+
+	By("checking cluster v4 or v6")
+	clusterOutput, err := f.CloudServices.EKS().DescribeCluster(f.Options.ClusterName)
+	Expect(err).NotTo(HaveOccurred())
+	isIPv4Cluster = false
+	if *clusterOutput.Cluster.KubernetesNetworkConfig.IpFamily == "ipv4" {
+		isIPv4Cluster = true
+	}
+	By("creating test namespace")
+	f.K8sResourceManagers.NamespaceManager().
+		CreateNamespace(utils.DefaultTestNamespace)
+
+	By(fmt.Sprintf("getting the node with the node label key %s and value %s",
+		f.Options.NgNameLabelKey, f.Options.NgNameLabelVal))
+	nodes, err := f.K8sResourceManagers.NodeManager().GetNodes(f.Options.NgNameLabelKey, f.Options.NgNameLabelVal)
+	Expect(err).ToNot(HaveOccurred())
+
+	By("verifying at least 1 node present for the test")
+	Expect(len(nodes.Items)).Should(BeNumerically(">", 0))
+
+	// Set the primary and secondary node for testing, these are used for a pod traffic test between two pods
+	for _, node := range nodes.Items {
+		if len(node.Spec.Taints) == 0 {
+			if primaryNode.Name == "" {
+				primaryNode = node
+				break
+			}
+		}
+	}
+	Expect(primaryNode.Name).To(Not(HaveLen(0)), "expected to find a non-tainted node")
+
+	// Get the node security group
+	instanceID := k8sUtils.GetInstanceIDFromNode(primaryNode)
+	primaryInstance, err := f.CloudServices.EC2().DescribeInstance(instanceID)
+	Expect(err).ToNot(HaveOccurred())
+
+	if isIPv4Cluster {
+		primaryNodeIp = primaryInstance.Ipv6Address
+	} else {
+		primaryNodeIp = primaryInstance.PublicIpAddress
+	}
+	Expect(primaryNodeIp).NotTo(BeNil())
+
+	By("getting the instance type from node label " + InstanceTypeNodeLabelKey)
+	instanceType := primaryNode.Labels[InstanceTypeNodeLabelKey]
+
+	By("getting the network interface details from ec2")
+	instanceOutput, err := f.CloudServices.EC2().DescribeInstanceType(instanceType)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Subtract 2 for coredns pods if any, both could be on same Interface
+	maxIPPerInterface = int(*instanceOutput[0].NetworkInfo.Ipv4AddressesPerInterface) - 2
```

Comment:
I think we can remove this for this test. It doesn't matter where coredns is running for these test cases, as they are testing egress from specific pods.

Reply:
The consideration of coreDNS is to calculate available IP in each interface. We goal is to have 2 ENI, and if we don't reduce it by 2 (which can be already used by coreDNS), we may end of having 3 ENI. 

For example, assuming `NetworkInfo.Ipv4AddressesPerInterface` is 15, and we use 15 for maxIPPerInterface, our deployment will be 2*15 = 30. 2 ENI is not enough since 2 IPs are used for coreDNS, and 3rd ENI has to be created.

Reply:
I think the test is resilient even if we have a 3rd ENI, right? The functions seem to tolerate any secondary ENI, not just one. But this looks fine anyway, as it makes sure we have two at the most